### PR TITLE
Feat.optionalcls

### PR DESCRIPTION
**Pull request by resoliwan**
move cls from island.cls mock to here to solve circular dependency.

### CODE REVIEW

File: src/cls.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/* tslint:disable */
+// Don't import any module! this file use hack!
+let ISLAND_USE_CLS = !(process.env.ISLAND_USE_CLS == 'false');
```
Comment (jiho-kr) [inline src/cls.ts]: `const` looks better

File: src/mock-cls.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
+/* tslint:disable */
+let namespaces = {};
```
Comment (jiho-kr) [inline src/mock-cls.ts]: const looks better

File: src/cls.ts (hunk 1 above)
Comment (UniverseHan) [reply src/cls.ts]: It looks bad library use env variable.

Comment (resoliwan) [reply src/cls.ts]: Yes. you are right!  It made dependency.

but we have to decide using mock cls or not before import other modules.

and it always return true unless env explicit set to false.

Do you have any better solution?

File: src/mock-cls.ts (hunk 2 above)
Comment (resoliwan) [reply src/mock-cls.ts]: When reset method is called, it assigns new value at namespace.

File: src/cls.ts (hunk 1 above)
Comment (UniverseHan) [reply src/cls.ts]: Hmm. No I don't have any idea.

**Pull request merged**